rust/crd/src/lib.rs

```diff
@@ -287,24 +331,34 @@ impl SparkApplication {
             format!("--conf spark.kubernetes.driver.container.image={}", self.spec.spark_image.as_ref().context(NoSparkImageSnafu)?),
             format!("--conf spark.kubernetes.executor.container.image={}", self.spec.spark_image.as_ref().context(NoSparkImageSnafu)?),
             format!("--conf spark.kubernetes.authenticate.driver.serviceAccountName={}", serviceaccount_name),
-        ];
+        ]);
 
         // See https://spark.apache.org/docs/latest/running-on-kubernetes.html#dependency-management
         // for possible S3 related properties
         if let Some(endpoint) = s3bucket.as_ref().and_then(|s3| s3.endpoint()) {
             submit_cmd.push(format!("--conf spark.hadoop.fs.s3a.endpoint={}", endpoint));
         }
-        if s3bucket.as_ref().and_then(|s3| s3.secret_class()).is_some() {
-            // We don't use the secret at all here, instead we assume the Self::env() has been
-            // called and this environment variables are availables.
-            submit_cmd.push(format!(
-                "--conf spark.hadoop.fs.s3a.access.key=${}",
-                ENV_AWS_ACCESS_KEY_ID
-            ));
-            submit_cmd.push(format!(
-                "--conf spark.hadoop.fs.s3a.secret.key=${}",
-                ENV_AWS_SECRET_ACCESS_KEY
-            ));
+
+        if let Some(conn) = s3bucket.as_ref().and_then(|i| i.connection.as_ref()) {
+            match conn.access_style {
+                Some(S3AccessStyle::Path) => {
+                    submit_cmd
+                        .push("--conf spark.hadoop.fs.s3a.path.style.access=true".to_string());
+                }
+                Some(S3AccessStyle::VirtualHosted) => {}
+                None => {}
+            }
+            if conn.credentials.as_ref().is_some() {
+                // We don't use the credentials at all here but assume they are available
+                submit_cmd.push(format!(
+                    "--conf spark.hadoop.fs.s3a.access.key=${}",
+                    ENV_AWS_ACCESS_KEY_ID
+                ));
+                submit_cmd.push(format!(
+                    "--conf spark.hadoop.fs.s3a.secret.key=${}",
+                    ENV_AWS_SECRET_ACCESS_KEY
+                ));
```

Comment:
Something like this (not tested)

```suggestion
                submit_cmd.push(format!(
                    "--conf spark.hadoop.fs.s3a.secret.key=$(cat {secret_dir}/{file_name})",
                env_var = ENV_AWS_SECRET_ACCESS_KEY,
                secret_dir = S3_SECRET_DIR_NAME,
                file_name = SECRET_ACCESS_KEY
            )
                ));
```

Reply:
Implemented as suggested and tested.